### PR TITLE
Refined the peer listener failed message in IC-PROXY mode

### PR DESCRIPTION
**Pull request by interma**
https://github.com/greenplum-db/gpdb/pull/16553 backported https://github.com/greenplum-db/gpdb/pull/16438 to 6x. And in it's code review process, got a suggestion to refine the error message: https://github.com/greenplum-db/gpdb/pull/16553#discussion_r1376929360

So, backport the enhancement of error message back to 7x.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by interma** [comment]
Dev pipeline is in https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-refine_report-error_icproxy-not-ready-rhel8, waiting it passed.

File: src/test/isolation2/expected/ic_proxy_listen_failed.out [hunk 1]
```diff
@@ -13,76 +13,90 @@ INSERT 0 100
...
+-- end_ignore
+(exited with code 0)
+!\retcode sleep 2 && gpstart -a;
+-- start_ignore
+20231110:15:29:55:591614 gpstart:gpdb:ubuntu-[INFO]:-Starting gpstart with args: -a
+20231110:15:29:55:591614 gpstart:gpdb:ubuntu-[INFO]:-Gathering information and validating the environment...
+20231110:15:29:55:591614 gpstart:gpdb:ubuntu-[INFO]:-Greenplum Binary Version: 'postgres (Greenplum Database) 7.0.0-alpha.0+dev.19326.ge2da67a1d1 build dev'
+20231110:15:29:55:591614 gpstart:gpdb:ubuntu-[INFO]:-Greenplum Catalog Version: '302307241'
+20231110:15:29:55:591614 gpstart:gpdb:ubuntu-[INFO]:-Starting Coordinator instance in admin mode
+20231110:15:29:55:591614 gpstart:gpdb:ubuntu-[INFO]:-CoordinatorStart pg_ctl cmd is env GPSESSID=0000000000 GPERA=None $GPHOME/bin/pg_ctl -D /home/ubuntu/gp/data7/master/gpseg-1 -l /home/ubuntu/gp/data7/master/gpseg-1/log/startup.log -w -t 600 -o " -c gp_role=utility " start
```
Comment (soumyadeep2007) [inline src/test/isolation2/expected/ic_proxy_listen_failed.out:52]: Stuff inside the ignore block can be ingored. There is no need to include changes caused due to the timestamps.

**Review by soumyadeep2007 — APPROVED** [review]
Please rewrite the commit message to be a bit more informative.
For e.g.:
`Refine the error messaging in proxy IC setup`

Thanks for taking care of this.

File: src/test/isolation2/expected/ic_proxy_listen_failed.out (hunk 1 above)
Comment (interma) [reply src/test/isolation2/expected/ic_proxy_listen_failed.out:52]: Got it, I will delete them.

**Comment by interma** [comment]
> Please rewrite the commit message to be a bit more informative.

Sure, I will squash my commit and refine the message when merging it.

**Pull request merged**